### DISCUSSION TITLE
QC for Additional Updates/Assessment Pipeline

### DISCUSSION THREAD

**Discussion by mattroumaya** [Gilead-Public/gsm#215] [General]
# QC for Additional Updates/Assessment Pipeline

- [ ] Documentation
    - [x] Function name captured in [Roxygen title](https://cran.r-project.org/web/packages/roxygen2/vignettes/rd.html#the-description-block)(e.g. "Adverse Event Assessment")
    - [x] Assessment Purpose captured in [Roxygen description](https://cran.r-project.org/web/packages/roxygen2/vignettes/rd.html#the-description-block) (e.g. "Evaluates site-level level Adverse Event Rates and flags rates that are abnormally high or low."
    - [ ] Input Data Requirements are described in the the [Roxygen details](https://cran.r-project.org/web/packages/roxygen2/vignettes/rd.html#the-description-block) under a Data section (`#' @details # Data Requirements`).
    - [ ] Statistical Assumptions - (Assess() and Analyze() only) List the statistical methods used and refer to detailed description in Analyze function in a the [Roxygen details](https://cran.r-project.org/web/packages/roxygen2/vignettes/rd.html#the-description-block) under a Statistics section (`#' @details # Statistical Assumptions`).
    - [ ] Parameters have detailed documentation using `@param`. Each parameter should include: parameter name, brief description, usage details, the default value (if any), is it required, list of valid options (if applicable)
    - [ ] All external dependencies are captured. Use `@importFrom` when importing 3 or fewer functions, and use `@import` otherwise. 
    - [ ] Output Data Standards are described under `@returns`
    - [x] At least 1 example is provided under `@examples`
- [ ] Error Checking
    - [ ]  Basic checks for all all parameters should be included using `stopifnot()` or similar logic. (e.g. `stopifnot("dfInput is not a dataframe"=is.data.frame(dfInput))`)
    - [ ] Tests confirm that `stopifnot()` parameter checks are working as expected. 
    - [ ] Tests confirm that the input data has required columns (if any)
    - [ ] Tests confirm that the output data has the expected columns
    - [ ] Tests confirm intended functionality for each parameter
    - [ ] Tests confirm that missing data in required columns is handled appropriately and errors/warnings are produced if needed. 
- [ ] Basic QC
    - [ ] Assessment has User Requirements + Qualification tests captured using {valtools} framework. Report is generating as expected and all checks are passing. 
    - [x] Code is well commented and easy to read
    - [x] No file paths or other company-specific data is present
    - [ ] Function calls from non-tidyverse dependencies are called via `::`
    - [x] devtools::check() passes with no errors/warnings/notes
    - [x] package docs up to date. devtools::document() doesn't change any files
    - [x] Uses tidyverse best practices for standard data manipulation (start discussion thread if unclear)
    - [x] All new dependencies add significant value (start discussion thread)
    - [x] All GitHub Actions run with no errors

**Comment by mattroumaya** [comment]
## Analyze_Chisq()

**Input Data Requirements are described in the the Roxygen details under a Data section (#' @details # Data Requirements).**
- Slight formatting changes needed - add _required_
  - Suggested fix: _The input data (`dfTransformed`) for Analyze_Chisq is typically created using \code{\link{Transform_EventCount}}  and should be one record per site with required columns for..._
- For input data, change [`Count`](https://github.com/Gilead-BioStats/gsm/blob/dev/R/Analyze_Chisq.R#L18) --> `TotalCount`

**Statistical Assumptions - (Assess() and Analyze() only) List the statistical methods used and refer to detailed description in Analyze function in a the Roxygen details under a Statistics section (#' @details # Statistical Assumptions).**
- Needs statistical assumptions section

**All external dependencies are captured. Use @importFrom when importing 3 or fewer functions, and use @import otherwise.**
- For [imports section](https://github.com/Gilead-BioStats/gsm/blob/dev/R/Analyze_Chisq.R#L24-L26)
  - Add `@import dplyr`
  - Add `@importFrom tidyr unnest`
  - Remove `as.formula` from `@importFrom stats`
  - Remove `map_df` from `@importFrom purrr`

**Basic checks for all all parameters should be included using stopifnot() or similar logic. (e.g. stopifnot("dfInput is not a dataframe"=is.data.frame(dfInput)))**
- `NA` handling for `SiteID` needed
- Check that `length(strOutcome) == 1`
- Add descriptions to `stopifnot()` errors

**Function calls from non-tidyverse dependencies are called via ::**
- [`chisq.test(SiteTable)` --> `stats::chisq.test(SiteTable)`](https://github.com/Gilead-BioStats/gsm/blob/dev/R/Analyze_Chisq.R#L54)
- [`map() --> purrr::map()`](https://github.com/Gilead-BioStats/gsm/blob/dev/R/Analyze_Chisq.R#L58-L59)
- [`unnest() --> tidyr::unnest()`](https://github.com/Gilead-BioStats/gsm/blob/dev/R/Analyze_Chisq.R#L60)

**Many QC checklist items cannot be checked due to the following**
🔴 No unit tests for `Analyze_Chisq()` - need to be created to pass **Error Checking** part of QC checklist.

---

## Analyze_Fisher()

**Input Data Requirements are described in the the Roxygen details under a Data section (#' @details # Data Requirements).**
- Slight formatting changes needed - add _required_
  - Suggested fix: _The input data (`dfTransformed`) for Analyze_Fisher is typically created using \code{\link{Transform_EventCount}}  and should be one record per site with required columns for..._
- For input data, change [`Count`](https://github.com/Gilead-BioStats/gsm/blob/dev/R/Analyze_Fisher.R#L18) --> `TotalCount`

**Statistical Assumptions - (Assess() and Analyze() only) List the statistical methods used and refer to detailed description in Analyze function in a the Roxygen details under a Statistics section (#' @details # Statistical Assumptions).**
- Needs statistical assumptions section

**All external dependencies are captured. Use @importFrom when importing 3 or fewer functions, and use @import otherwise.**
- For [imports section](https://github.com/Gilead-BioStats/gsm/blob/dev/R/Analyze_Fisher.R#L24-L27)
  - Add `@import dplyr`
  - Remove `as.formula` from `@importFrom stats`

**Basic checks for all all parameters should be included using stopifnot() or similar logic. (e.g. stopifnot("dfInput is not a dataframe"=is.data.frame(dfInput)))**
- `NA` handling for `SiteID` needed
- Check that `length(strOutcome) == 1`
- Add descriptions to `stopifnot()` errors

**Function calls from non-tidyverse dependencies are called via ::**
- [`unnest() --> tidyr::unnest()](https://github.com/Gilead-BioStats/gsm/blob/dev/R/Analyze_Fisher.R#L61)

**Many QC checklist items cannot be checked due to the following**
🔴 No unit tests for `Analyze_Fisher()` - need to be created to pass **Error Checking** part of QC checklist.

## Analyze_Poisson()

**Input Data Requirements are described in the the Roxygen details under a Data section (#' @details # Data Requirements).**
- Slight formatting changes needed - add _required_
  - Suggested fix: _The input data (`dfTransformed`) for Analyze_Poisson is typically created using \code{\link{Transform_EventCount}} and should be one record per site with required columns for..._
- For input data, [`dfTransformed`](https://github.com/Gilead-BioStats/gsm/blob/dev/R/Analyze_Poisson.R#L15-L19) should contain the columns:
  - `SiteID - Site ID`
  - `N - Number of participants`
  - `TotalCount - Number of events`
  - `TotalExposure - Number of days exposure`
  - `Rate - Rate of exposure (?)`

**Statistical Assumptions - (Assess() and Analyze() only) List the statistical methods used and refer to detailed description in Analyze function in a the Roxygen details under a Statistics section (#' @details # Statistical Assumptions).**
- Needs statistical assumptions section

**Parameters have detailed documentation using @param. Each parameter should include: parameter name, brief description, usage details, the default value (if any), is it required, list of valid options (if applicable)**
- Remove ["Must include"](https://github.com/Gilead-BioStats/gsm/blob/dev/R/Analyze_Poisson.R#L21) from the end of `@param` description

**All external dependencies are captured. Use @importFrom when importing 3 or fewer functions, and use @import otherwise.**
- For [imports section](https://github.com/Gilead-BioStats/gsm/blob/dev/R/Analyze_Poisson.R#L23-L24)
  - Add `@import dplyr`

**Output Data Standards are described under @returns**
- Returns expected columns, but is also returning additional columns: `.std.resid, .hat, .sigma, .cooksd`
  - Is this expected behavior?

**Basic checks for all all parameters should be included using stopifnot() or similar logic. (e.g. stopifnot("dfInput is not a dataframe"=is.data.frame(dfInput)))**
- `NA` handling for `SiteID` needed
- Add descriptions to `stopifnot()` errors

**Tests confirm that stopifnot() parameter checks are working as expected.**
- Add tests to cover all required columns in `dfInput` - add to [lines 19 - 22](https://github.com/Gilead-BioStats/gsm/blob/dev/tests/testthat/test_Analyze_Poisson.R#L19-L22)
- Currently uses `safetyData` for unit tests

**Tests confirm that the input data has required columns (if any)**
- Need test to confirm input data contains columns: `SubjectID`, `SiteID`, `TotalCount`, `TotalExposure`, `Rate`

**Tests confirm that the output data has the expected columns**
- Need to add test for expected output column names


**Tests confirm that missing data in required columns is handled appropriately and errors/warnings are produced if needed.**
- Need tests for NA values
---
## Analyze_Wilcoxon()

**Input Data Requirements are described in the the Roxygen details under a Data section (#' @details # Data Requirements).**
- Slight formatting changes needed - add _required_
  - Suggested fix: _The input data (`dfTransformed`) for Analyze_Wilcoxon is typically created using \code{\link{Transform_EventCount}}  and should be one record per site with required columns for..._
- For input data, [`dfTransformed`](https://github.com/Gilead-BioStats/gsm/blob/dev/R/Analyze_Wilcoxon.R#L15-L19) should contain the columns:
  - `SiteID - Site ID`
  - `N - Number of participants`
  - `TotalCount - Number of events`
  - `TotalExposure - Number of days exposure`
  - `Rate - Rate of exposure (?)`

**Statistical Assumptions - (Assess() and Analyze() only) List the statistical methods used and refer to detailed description in Analyze function in a the Roxygen details under a Statistics section (#' @details # Statistical Assumptions).**
- Needs statistical assumptions section

**All external dependencies are captured. Use @importFrom when importing 3 or fewer functions, and use @import otherwise.**
- Add `@import dplyr`
- Remove `map_df` from `@importFrom purrr`

**Output Data Standards are described under @returns**
- Need to update `@return` section:
  - Suggested fix: _data.frame with one row per site, columns:   SiteID, N , TotalCount, TotalExposure, Rate, Estimate, Pvalue_

**Basic checks for all all parameters should be included using stopifnot() or similar logic. (e.g. stopifnot("dfInput is not a dataframe"=is.data.frame(dfInput)))**
- `NA` handling for `SiteID` needed
- Check that `length(strOutcome) == 1`
- Add descriptions to `stopifnot()` errors

**Tests confirm that stopifnot() parameter checks are working as expected.**
- Add [test](https://github.com/Gilead-BioStats/gsm/blob/dev/tests/testthat/test_Analyze_Wilcoxon.R) to cover `strOutcome %in% names()`
- Currently uses `safetyData` for unit tests

**Tests confirm that the input data has required columns (if any)**
- Need test to confirm input data contains columns: `SiteID, N, Rate`

**Tests confirm intended functionality for each parameter**
- Need test for `strOutcome`

**Tests confirm that missing data in required columns is handled appropriately and errors/warnings are produced if needed.**
- Need tests for `NA` values

**Function calls from non-tidyverse dependencies are called via ::**
- [`wilcox.test() --> stats::wilcox.test()`](https://github.com/Gilead-BioStats/gsm/blob/dev/R/Analyze_Wilcoxon.R#L48)
- [`unnest() --> tidyr::unnest()`](https://github.com/Gilead-BioStats/gsm/blob/dev/R/Analyze_Wilcoxon.R#L54)

---
## Flag()

**Parameters have detailed documentation using @param. Each parameter should include: parameter name, brief description, usage details, the default value (if any), is it required, list of valid options (if applicable)**
- Minor fix: clean up capitalization and punctuation for `@params`

**Basic checks for all all parameters should be included using stopifnot() or similar logic. (e.g. stopifnot("dfInput is not a dataframe"=is.data.frame(dfInput)))**
- Check that `SiteID %in% names(dfAnalyzed)`
- Check that `length(strColumn) == 1`
- Add descriptions to `stopifnot()` errors

**Tests confirm that stopifnot() parameter checks are working as expected.**
- Add test to confirm `SiteID` is in input
- Currently uses `safetyData` for unit tests

**Tests confirm that the input data has required columns (if any)**
- Need test to confirm input data contains columns: `SiteID, strColumn (default = "Pvalue")`

**Tests confirm that the output data has the expected columns**
- Need to add explicit [test](https://github.com/Gilead-BioStats/gsm/blob/dev/tests/testthat/test_Flag.R) for expected output column names

---

## Summarize()

**Input Data Requirements are described in the the Roxygen details under a Data section (#' @details # Data Requirements).**
- Needs a [data specification section](https://github.com/Gilead-BioStats/gsm/blob/dev/R/Summarize.R#L1-L20)

**Output Data Standards are described under @returns**
- Need to update [`@return`](https://github.com/Gilead-BioStats/gsm/blob/dev/R/Summarize.R#L9) section
  - Suggested fix: _Simplified finding data frame with columns: Assessment, Label, SiteID, N, Pvalue, Flag_

**Basic checks for all all parameters should be included using stopifnot() or similar logic. (e.g. stopifnot("dfInput is not a dataframe"=is.data.frame(dfInput)))**
- Check that `length(cAssessment) == 1`
- Check that `length(cLabel) == 1`
- Add descriptions to `stopifnot()` errors

**Tests confirm that stopifnot() parameter checks are working as expected.**
- Add tests for specific inputs throwing error ([lines 17-21](https://github.com/Gilead-BioStats/gsm/blob/dev/tests/testthat/test_Summarize.R#L17-L21), add specific tests for `cCountCol` and `cExposureCol`)

**Tests confirm that the input data has required columns (if any)**
- Need test to confirm input data contains `"SiteID", "N", "PValue", "Flag"`

**Tests confirm that the output data has the expected columns**
- Need to add explicit test for expected output column names

**Tests confirm that missing data in required columns is handled appropriately and errors/warnings are produced if needed.**
- Need tests for `NA` values

---

## Transform_EventCount()

**Parameters have detailed documentation using @param. Each parameter should include: parameter name, brief description, usage details, the default value (if any), is it required, list of valid options (if applicable)**
- Minor fix: clean up capitalization and punctuation for `@params`

**All external dependencies are captured. Use @importFrom when importing 3 or fewer functions, and use @import otherwise.**
- Add `@import dplyr`

**Basic checks for all all parameters should be included using stopifnot() or similar logic. (e.g. stopifnot("dfInput is not a dataframe"=is.data.frame(dfInput)))**
- Check that `SiteID %in% dfInput`

**Tests confirm that stopifnot() parameter checks are working as expected.**
- Currently uses `safetyData` for unit tests

**Tests confirm that the input data has required columns (if any)**
- Need test to confirm input data contains columns: `SiteID, Count`

**Tests confirm that the output data has the expected columns**
- Need to add explicit test for expected output column names

@jwildfire - ready for your review! 🎉

**Reply by jwildfire** [reply]
> ## Analyze_Chisq()
> **Input Data Requirements are described in the the Roxygen details under a Data section (#' @details # Data Requirements).**
> 
> * Slight formatting changes needed - add _required_
>   
>   * Suggested fix: _The input data (`dfTransformed`) for Analyze_Chisq is typically created using \code{\link{Transform_EventCount}}  and should be one record per site with required columns for..._
> * For input data, change [`Count`](https://github.com/Gilead-BioStats/gsm/blob/dev/R/Analyze_Chisq.R#L18) --> `TotalCount`
> 
> **Statistical Assumptions - (Assess() and Analyze() only) List the statistical methods used and refer to detailed description in Analyze function in a the Roxygen details under a Statistics section (#' @details # Statistical Assumptions).**
> 
> * Needs statistical assumptions section
> 
> **All external dependencies are captured. Use @importFrom when importing 3 or fewer functions, and use @import otherwise.**
> 
> * For [imports section](https://github.com/Gilead-BioStats/gsm/blob/dev/R/Analyze_Chisq.R#L24-L26)
>   
>   * Add `@import dplyr`
>   * Add `@importFrom tidyr unnest`
>   * Remove `as.formula` from `@importFrom stats`
>   * Remove `map_df` from `@importFrom purrr`
> 
> **Basic checks for all all parameters should be included using stopifnot() or similar logic. (e.g. stopifnot("dfInput is not a dataframe"=is.data.frame(dfInput)))**
> 
> * `NA` handling for `SiteID` needed
> * Check that `length(strOutcome) == 1`
> * Add descriptions to `stopifnot()` errors
> 
> **Function calls from non-tidyverse dependencies are called via ::**
> 
> * [`chisq.test(SiteTable)` --> `stats::chisq.test(SiteTable)`](https://github.com/Gilead-BioStats/gsm/blob/dev/R/Analyze_Chisq.R#L54)
> * [`map() --> purrr::map()`](https://github.com/Gilead-BioStats/gsm/blob/dev/R/Analyze_Chisq.R#L58-L59)
> * [`unnest() --> tidyr::unnest()`](https://github.com/Gilead-BioStats/gsm/blob/dev/R/Analyze_Chisq.R#L60)
> 
> **Many QC checklist items cannot be checked due to the following** 

🔴 No unit tests for `Analyze_Chisq()` - need to be created to pass **Error Checking** part of QC checklist.


👍  Lets file 2 issues one for the unit tests and one for everything else.

**Reply by jwildfire** [reply]
> ## Analyze_Fisher()
> **Input Data Requirements are described in the the Roxygen details under a Data section (#' @details # Data Requirements).**
> 
> * Slight formatting changes needed - add _required_
>   
>   * Suggested fix: _The input data (`dfTransformed`) for Analyze_Fisher is typically created using \code{\link{Transform_EventCount}}  and should be one record per site with required columns for..._
> * For input data, change [`Count`](https://github.com/Gilead-BioStats/gsm/blob/dev/R/Analyze_Fisher.R#L18) --> `TotalCount`
> 
> **Statistical Assumptions - (Assess() and Analyze() only) List the statistical methods used and refer to detailed description in Analyze function in a the Roxygen details under a Statistics section (#' @details # Statistical Assumptions).**
> 
> * Needs statistical assumptions section
> 
> **All external dependencies are captured. Use @importFrom when importing 3 or fewer functions, and use @import otherwise.**
> 
> * For [imports section](https://github.com/Gilead-BioStats/gsm/blob/dev/R/Analyze_Fisher.R#L24-L27)
>   
>   * Add `@import dplyr`
>   * Remove `as.formula` from `@importFrom stats`
> 
> **Basic checks for all all parameters should be included using stopifnot() or similar logic. (e.g. stopifnot("dfInput is not a dataframe"=is.data.frame(dfInput)))**
> 
> * `NA` handling for `SiteID` needed
> * Check that `length(strOutcome) == 1`
> * Add descriptions to `stopifnot()` errors
> 
> **Function calls from non-tidyverse dependencies are called via ::**
> 
> * [`unnest() --> tidyr::unnest()](https://github.com/Gilead-BioStats/gsm/blob/dev/R/Analyze_Fisher.R#L61)
> 
> **Many QC checklist items cannot be checked due to the following** 🔴 No unit tests for `Analyze_Fisher()` - need to be created to pass **Error Checking** part of QC checklist.

👍  good. Let's create 2 issues here as well.

**Reply by jwildfire** [reply]
> ## Analyze_Poisson()
> **Input Data Requirements are described in the the Roxygen details under a Data section (#' @details # Data Requirements).**
> 
> * Slight formatting changes needed - add _required_
>   
>   * Suggested fix: _The input data (`dfTransformed`) for Analyze_Poisson is typically created using \code{\link{Transform_EventCount}} and should be one record per site with required columns for..._
> * For input data, [`dfTransformed`](https://github.com/Gilead-BioStats/gsm/blob/dev/R/Analyze_Poisson.R#L15-L19) should contain the columns:
>   
>   * `SiteID - Site ID`
>   * `N - Number of participants`
>   * `TotalCount - Number of events`
>   * `TotalExposure - Number of days exposure`
>   * `Rate - Rate of exposure (?)`
> 
> **Statistical Assumptions - (Assess() and Analyze() only) List the statistical methods used and refer to detailed description in Analyze function in a the Roxygen details under a Statistics section (#' @details # Statistical Assumptions).**
> 
> * Needs statistical assumptions section
> 
> **Parameters have detailed documentation using @param. Each parameter should include: parameter name, brief description, usage details, the default value (if any), is it required, list of valid options (if applicable)**
> 
> * Remove ["Must include"](https://github.com/Gilead-BioStats/gsm/blob/dev/R/Analyze_Poisson.R#L21) from the end of `@param` description
> 
> **All external dependencies are captured. Use @importFrom when importing 3 or fewer functions, and use @import otherwise.**
> 
> * For [imports section](https://github.com/Gilead-BioStats/gsm/blob/dev/R/Analyze_Poisson.R#L23-L24)
>   
>   * Add `@import dplyr`
> 
> **Output Data Standards are described under @returns**
> 
> * Returns expected columns, but is also returning additional columns: `.std.resid, .hat, .sigma, .cooksd`
>   
>   * Is this expected behavior?
> 
> **Basic checks for all all parameters should be included using stopifnot() or similar logic. (e.g. stopifnot("dfInput is not a dataframe"=is.data.frame(dfInput)))**
> 
> * `NA` handling for `SiteID` needed
> * Add descriptions to `stopifnot()` errors
> 
> **Tests confirm that stopifnot() parameter checks are working as expected.**
> 
> * Add tests to cover all required columns in `dfInput` - add to [lines 19 - 22](https://github.com/Gilead-BioStats/gsm/blob/dev/tests/testthat/test_Analyze_Poisson.R#L19-L22)
> * Currently uses `safetyData` for unit tests
> 
> **Tests confirm that the input data has required columns (if any)**
> 
> * Need test to confirm input data contains columns: `SubjectID`, `SiteID`, `TotalCount`, `TotalExposure`, `Rate`
> 
> **Tests confirm that the output data has the expected columns**
> 
> * Need to add test for expected output column names
> 
> **Tests confirm that missing data in required columns is handled appropriately and errors/warnings are produced if needed.**
> 
> * Need tests for NA values

👍  Some of this has been addressed in #213, but go ahead and file an issue with everything and we can just cross some things off after that PR is merged.

**Reply by jwildfire** [reply]
> ## Analyze_Wilcoxon()
> **Input Data Requirements are described in the the Roxygen details under a Data section (#' @details # Data Requirements).**
> 
> * Slight formatting changes needed - add _required_
>   
>   * Suggested fix: _The input data (`dfTransformed`) for Analyze_Wilcoxon is typically created using \code{\link{Transform_EventCount}}  and should be one record per site with required columns for..._
> * For input data, [`dfTransformed`](https://github.com/Gilead-BioStats/gsm/blob/dev/R/Analyze_Wilcoxon.R#L15-L19) should contain the columns:
>   
>   * `SiteID - Site ID`
>   * `N - Number of participants`
>   * `TotalCount - Number of events`
>   * `TotalExposure - Number of days exposure`
>   * `Rate - Rate of exposure (?)`
> 
> **Statistical Assumptions - (Assess() and Analyze() only) List the statistical methods used and refer to detailed description in Analyze function in a the Roxygen details under a Statistics section (#' @details # Statistical Assumptions).**
> 
> * Needs statistical assumptions section
> 
> **All external dependencies are captured. Use @importFrom when importing 3 or fewer functions, and use @import otherwise.**
> 
> * Add `@import dplyr`
> * Remove `map_df` from `@importFrom purrr`
> 
> **Output Data Standards are described under @returns**
> 
> * Need to update `@return` section:
>   
>   * Suggested fix: _data.frame with one row per site, columns:   SiteID, N , TotalCount, TotalExposure, Rate, Estimate, Pvalue_
> 
> **Basic checks for all all parameters should be included using stopifnot() or similar logic. (e.g. stopifnot("dfInput is not a dataframe"=is.data.frame(dfInput)))**
> 
> * `NA` handling for `SiteID` needed
> * Check that `length(strOutcome) == 1`
> * Add descriptions to `stopifnot()` errors
> 
> **Tests confirm that stopifnot() parameter checks are working as expected.**
> 
> * Add [test](https://github.com/Gilead-BioStats/gsm/blob/dev/tests/testthat/test_Analyze_Wilcoxon.R) to cover `strOutcome %in% names()`
> * Currently uses `safetyData` for unit tests
> 
> **Tests confirm that the input data has required columns (if any)**
> 
> * Need test to confirm input data contains columns: `SiteID, N, Rate`
> 
> **Tests confirm intended functionality for each parameter**
> 
> * Need test for `strOutcome`
> 
> **Tests confirm that missing data in required columns is handled appropriately and errors/warnings are produced if needed.**
> 
> * Need tests for `NA` values
> 
> **Function calls from non-tidyverse dependencies are called via ::**
> 
> * [`wilcox.test() --> stats::wilcox.test()`](https://github.com/Gilead-BioStats/gsm/blob/dev/R/Analyze_Wilcoxon.R#L48)
> * [`unnest() --> tidyr::unnest()`](https://github.com/Gilead-BioStats/gsm/blob/dev/R/Analyze_Wilcoxon.R#L54)

👍  looks great. 1 issue for everything.

**Reply by jwildfire** [reply]
The  notes for `Flag()`, `Summarize()` and `Transform_EventCount()` all make sense to me as well. Go ahead with 1 issue for each function.
